### PR TITLE
improve readability on important methods base mailer

### PR DESCRIPTION
**Pull request by DavidNV**
Just some minor changes to ease the reading of important methods of actionmailer/lib/base.rb file. Plus, the check on args could be a nil or a value so I added the present method to check it.

### CODE REVIEW

File: actionmailer/lib/action_mailer/base.rb [hunk 1]
```diff
@@ -574,11 +574,7 @@ def set_payload_for_mail(payload, mail)
       end
 
       def method_missing(method_name, *args)
-        if action_methods.include?(method_name.to_s)
-          MessageDelivery.new(self, method_name, *args)
-        else
-          super
-        end
+        action_methods.include?(method_name.to_s) ? MessageDelivery.new(self, method_name, *args) : super
```
Comment (rafaelfranca) [inline actionmailer/lib/action_mailer/base.rb]: This is not improving the readability. Could you keep the if/else?

File: actionmailer/lib/action_mailer/base.rb [hunk 2]
```diff
@@ -664,11 +660,7 @@ def mailer_name
     # +nil+ in order to reset the value otherwise another field will be added
     # for the same header.
     def headers(args = nil)
-      if args
-        @_message.headers(args)
-      else
-        @_message
-      end
+      args.present? ? @_message.headers(args) : @_message
```
Comment (rafaelfranca) [inline actionmailer/lib/action_mailer/base.rb]: Same here about the if/else

Comment (rafaelfranca) [inline actionmailer/lib/action_mailer/base.rb]: Why is this checking for `present?` now? Do we want to handle `""` as same as `nil`? I don't think so.

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: actionmailer/lib/action_mailer/base.rb (hunk 2 above)
Comment (DavidNV) [inline actionmailer/lib/action_mailer/base.rb]: Go it. I will close the PR, Thank you for your feedback.

**Pull request closed without merging**